Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -212,6 +218,24 @@ public class ApolloCodegen {
       throw Error.schemaNameConflict(name: context.schemaNamespace)
     }
   }
+  
+  /// Validates that there are no type conflicts within a SelectionSet
+  static private func validateTypeConflicts(for selectionSet: IR.SelectionSet, with context: ConfigurationContext) throws {
+    // Check for type conflicts resulting from singularization/pluralization of fields
+    var typeNames = [String: String]()
+    var fields: [IR.EntityField] = selectionSet.selections.direct?.fields.values.compactMap({ $0 as? IR.EntityField }) ?? []
+    fields.append(contentsOf: selectionSet.selections.merged.fields.values.compactMap({ $0 as? IR.EntityField }))
+
+    try fields.forEach({ field in
```

Comment:
I'm not opposed to using `forEach` but I'm not sure what the value of that over `for field in fields` is here.

Nit: To better follow swift standard style guidelines, you can omit the `()` here. `try fields.forEach { ... }`

Reply:
Honestly was just a style preference, happy to change it to `for field in fields` if we want to enforce that style for this type of loop.

Reply:
Really doesn't matter much to me. I have always preferred `for ... in ...` but it's just a personal preference! It was just odd for me to see that! Go ahead and leave it. :)